oasislmf/preparation/gul_inputs.py

```diff
@@ -148,35 +198,12 @@ def get_gul_input_items(
     # Remove any duplicate column names used to assign group_id
     group_id_cols = list(set(group_id_cols))
 
-    # Ignore any column names used to assign group_id that are missing or not supported
-    # Valid group id columns can be either
-    # 1. exist in the location file
-    # 2. be listed as a useful internal col
-    valid_oasis_group_cols = [
-        'item_id',
-        'peril_id',
-        'coverage_id',
-        'coverage_type_id',
-        'peril_correlation_group'
-    ]
-    for col in group_id_cols:
-        if col not in list(exposure_df.columns) + valid_oasis_group_cols:
-            warnings.warn('Column {} not found in loc file, or a valid internal oasis column'.format(col))
-            group_id_cols.remove(col)
-
-    # here we check to see if the correlation file is here, if it is then we need to add the "peril_correlation_group" to the valid_oasis_group_cols
-    peril_correlation_group = 'peril_correlation_group'
-    correlations_files = [
-        f"{output_dir}/correlations.csv",
-        f"{output_dir}/correlations.bin",
-    ]
-    for file_path in correlations_files:
-        if os.path.exists(path=file_path):
-            if peril_correlation_group not in group_id_cols:
-                group_id_cols.append(peril_correlation_group)
-            break
-
-
+    # it is assumed that correlations are False for now, correlations for group ID hashing are assessed later on in
```

Comment:
to avoid hashing twice, you could do the merge with the correlation data in this function instead of doing it after calling it. That should remove your chicken and egg problem.

Reply:
hashing is only performed once now